samples/src/test/java/org/autorefactor/refactoring/rules/samples_out/StringSample.java

```diff
@@ -51,21 +51,21 @@ public String replaceStringToString(String s) {
     }
 
     public String replaceToStringCallInStringConcat() {
-        String s1 = "" + Boolean.TRUE + Boolean.FALSE;
-        String s2 = Boolean.TRUE + "" + Boolean.FALSE;
+        String s1 = " " + Boolean.TRUE + Boolean.FALSE;
+        String s2 = Boolean.TRUE + " " + Boolean.FALSE;
         return s1 + s2;
     }
 
     public String replaceToStringCallOnCurrentObjectInStringConcat() {
-        String s1 = "" + this + this;
-        String s2 = this + "" + this;
+        String s1 = " " + this + this;
+        String s2 = this + " " + this;
         return s1 + s2;
     }
 
     public class TestWithQualifiedToString {
         public String replaceToStringCallOnCurrentObjectInStringConcat() {
-            String s1 = "" + StringSample.this + StringSample.this + this;
-            String s2 = StringSample.this + "" + this;
+            String s1 = " " + StringSample.this + StringSample.this + this;
+            String s2 = StringSample.this + " " + this;

```

Comment:
I disagree with the changes in this section.
Please revert them.

They highlight the fact that this code is missing one check on `extendedOperands(InfixExpression).isEmpty()`.